### DISCUSSION TITLE
How to go from the example test app to production?

### DISCUSSION THREAD

**Discussion by niyonx** [flutter-stripe/flutter_stripe#471] [Q&A]
Does changing the keys from test to live keys all I have to do? How does the server get replaced then?

An example flow would be to 

1. Create a customer and have its id (rest api)
2. List his cards/ create new cards (rest api)
3. Then do the payment sheet flow

Please help, much appreciated 🙏

**Comment by remonh87** [comment]
is switching the keys enough?

you need to replace your test keys with production keys. You also need to add the production publishable key to your app set test mode to false in case you use the payment sheet.
should i host the yarn server even when using the live keys?

Your backend needs to be reachable for the app that you are using so test you need to host it.
can I use the live keys without the yarn server?

you always need a backend because the backend is responsible for creating the payment intents. Depending on the payment methods that you provide you should check the stripe docs on what to implement server side. Our library only handles the client side.
the amount to be charged in the example is 10.99, how do I change the amount from the flutter side?

You can't and that would be security wise very bad. What you need to do is request a new payment intent (with the new charge) in case the amount is wrong. Your backend needs to handle this with Stripe.
Hope this helps.
